jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/LambdaSamRunConfigurationProducer.kt

```diff
@@ -20,37 +22,53 @@ import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
 class LambdaSamRunConfigurationProducer : RunConfigurationProducer<SamRunConfiguration>(getFactory()) {
     override fun getConfigurationSettingsList(runManager: RunManager): List<RunnerAndConfigurationSettings> {
         // Filter all Lambda run CONFIGURATIONS down to only ones that are Lambda SAM for this run producer
-        return super.getConfigurationSettingsList(runManager)
-                .filter { it.configuration is SamRunConfiguration }
+        return super.getConfigurationSettingsList(runManager).filter { it.configuration is SamRunConfiguration }
     }
 
-    override fun setupConfigurationFromContext(
-        configuration: SamRunConfiguration,
-        context: ConfigurationContext,
-        sourceElement: Ref<PsiElement>
-    ): Boolean {
+    override fun setupConfigurationFromContext(configuration: SamRunConfiguration, context: ConfigurationContext, sourceElement: Ref<PsiElement>): Boolean {
         val element = context.psiLocation ?: return false
+        val parent = element.parent
```

Comment:
Why does setup require the parent, yet isConfigfromContext doesnt?

Are the contexts not equal?

Reply:
good catch - added a test and fixed the logic.